src/main/java/vg/civcraft/mc/civmodcore/CoreConfigManager.java

```diff
@@ -22,57 +31,186 @@ public CoreConfigManager(ACivMod plugin) {
 		this.logger = plugin.getLogger();
 	}
 
-	public boolean parse() {
-		plugin.info("Parsing config file of " + plugin.getName());
-		plugin.saveDefaultConfig();
-		plugin.reloadConfig();
-		FileConfiguration config = plugin.getConfig();
-		debug = config.getBoolean("debug", false);
-		logReplies = config.getBoolean("logReplies", false);
+	/**
+	 * Parses this civ plugin's config. It will also save a default config (if one is not already present) and reload
+	 * the config, so there's no need to do so yourself beforehand.
+	 *
+	 * @return Returns true if the config was successfully parsed.
+	 */
+	public final boolean parse() {
+		this.plugin.info(ChatColor.BLUE + "Parsing config.");
+		this.plugin.saveDefaultConfig();
+		this.plugin.reloadConfig();
+		FileConfiguration config = this.plugin.getConfig();
+		// Parse debug value
+		this.debug = config.getBoolean("debug", false);
+		this.plugin.info("Debug mode: " + (this.debug ? "enabled" : "disabled"));
+		// Parse reply logging value
+		this.logReplies = config.getBoolean("logReplies", false);
+		this.plugin.info("Logging replies: " + (this.logReplies ? "enabled" : "disabled"));
+		// Allow child class parsing
 		boolean worked = parseInternal(config);
 		if (worked) {
-			plugin.info("Successfully parsed config file of " + plugin.getName());
-		} else {
-			plugin.info("Failed to parse config file of " + plugin.getName() + ". Errors were encountered");
+			plugin.info(ChatColor.BLUE + "Config parsed.");
+		}
+		else {
+			plugin.warning("Failed to pass config!");
 		}
 		return worked;
 	}
 
-	public boolean isDebugEnabled() {
-		return debug;
+	/**
+	 * An internal parser method intended to be overriden by child classes.
+	 *
+	 * @param config The root config section.
+	 * @return Return true if the
+	 */
+	protected boolean parseInternal(ConfigurationSection config) {
```

Comment:
Why should this not be abstract, it makes using the class properly harder

Reply:
Because having the method be abstract requires the class to be abstract, which means you're then requiring developers who are content with the functionality CoreConfigManager provides, to create a useless, empty extension class. I put notes in the commits.